public.tf

```diff
@@ -0,0 +1,74 @@
+module "public_label" {
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.2.2"
+  namespace  = "${var.namespace}"
+  name       = "${var.availability_zone}"
+  attributes = ["public"]
+  stage      = "${var.stage}"
+  delimiter  = "${var.delimiter}"
+  tags       = "${var.tags}"
+}
+
+resource "aws_subnet" "public" {
+  count             = "${var.type == "public" ? length(var.names) : 0}"
+  vpc_id            = "${var.vpc_id}"
+  availability_zone = "${var.availability_zone}"
+  cidr_block        = "${cidrsubnet(var.cidr_block, ceil(log(var.max_subnets_number, 2)), count.index)}"
```

Comment:
I think in this case need some additional check, because  length(var.names) can be larger than`var.max_subnets_number`, something like : 
```terraform
locals {
 subnets_count = "${length(var.names) > var.max_subnets_number ?  length(var.names) : var.max_subnets_number}"
}

resource "aws_subnet" "public" {
cidr_block        = "${cidrsubnet(var.cidr_block, ceil(log(local.subnets_count, 2)), count.index)}"
}

```

Reply:
Your suggestion is great!
But the goal is to prevent subnet CIDR reassignment.
In your case, if we request more subnets, than `max_subnet_number`, all subnet CIDRs will be changed and all existing subnets will be deleted.

Reply:
hmmm... I think the situation when `length(var.names)` larger than `var.max_subnets_number` will break the module

Reply:
I'll cause error, yes. But it will not cause chain of CIDR reassignments.
Imagine, if we have live system with 16 subnets, and when we create 17th subnet, all existing subnets will be recreated.

Reply:
yea, your and my variants have the vulnerability :) , one way fix it's some condition for force break of the loop, but with terraform it's impossible :)

Reply:
Previously, without `max_subnets_number`, subnet CIDRs were calculated dynamically, based `names` length.
So, if we need this capability, we can reset 2 last commits.
But since we want to `prevent` CIDRs reassignment, we've added `max_subnets_number`.
And we want to be sure, that nothing can cause mass CIDRs reassignment.

Your suggestion is good, and it has been implemented already.
But it was deprecated in favor of subnets structure stability.

Reply:
So, the `vulnerability`, actually, is a `foolproof`.